api/client.go

```diff
@@ -71,7 +74,12 @@ func (c *Client) Do(req *http.Request, v interface{}) (*http.Response, error) {
 	default:
 		if v != nil {
 			defer res.Body.Close()
-			if err := json.NewDecoder(res.Body).Decode(v); err != nil {
+
+			var bodyCopy bytes.Buffer
+			body := io.TeeReader(res.Body, &bodyCopy)
+
+			if err := json.NewDecoder(body).Decode(v); err != nil {
+				debug.Println("res.Body:", bodyCopy.String())
 				return nil, fmt.Errorf("error parsing API response - %s", err)
```

Comment:
Fetching without an api key now just fetches the exercise. :+1: 

Getting an error response from the server, now looks like this:
```
> ./cli fetch go leapa
verbose logging enabled
NewRequest: GET http://x.exercism.io/v2/exercises/go/leapa
StatusCode: 404
res.Body: {"error":"We don't have problem 'leapa' in 'go'"}
2015/11/21 11:16:59 unable to fetch problems (HTTP: 404) - We don't have problem 'leapa' in 'go'
```
We are displaying the error provided by the json response. Were any other changes required?

I'm wondering if we should always show the body response in verbose mode regardless of whether we have an error or not? It does add a lot of noise, but could be useful as we make changes to the erecism.io api. It could also be useful if, for example, someone has a proxy server that is mangling the response.